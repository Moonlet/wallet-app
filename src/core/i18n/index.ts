export * from './translation/translate-component';
export * from './translation/translate';
