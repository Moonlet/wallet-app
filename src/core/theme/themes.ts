export enum Theme {
    dark = 'dark',
    light = 'light'
}
