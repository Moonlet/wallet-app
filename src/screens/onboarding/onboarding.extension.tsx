import React from 'react';
import { View } from 'react-native';
import { Text } from '../../library';
import { NavigationActions } from 'react-navigation';
import stylesProvider from './styles-web';
import { withTheme, IThemeProps } from '../../core/theme/with-theme';
import { smartConnect } from '../../core/utils/smart-connect';
import { translate } from '../../core/i18n';
import { withNavigationParams, INavigationProps } from '../../navigation/with-navigation-params';
import QRCode from 'qrcode';
import { ConnectExtensionWeb } from '../../core/connect-extension/connect-extension-web.extension';

const navigationOptions = () => ({ headerShown: false });

export class OnboardingScreenComponent extends React.Component<
    INavigationProps & IThemeProps<ReturnType<typeof stylesProvider>>
> {
    public static navigationOptions = navigationOptions;
    public qrCanvas: HTMLCanvasElement;

    public async componentDidMount() {
        try {
            if (await ConnectExtensionWeb.isConnected()) {
                this.props.navigation.navigate(
                    'MainNavigation',
                    {},
                    NavigationActions.navigate({ routeName: 'Dashboard' })
                );
            } else {
                const res = await ConnectExtensionWeb.generateQRCodeUri();
                QRCode.toCanvas(this.qrCanvas, res.uri, { errorCorrectionLevel: 'H' });
                ConnectExtensionWeb.listenLastSyncForConnect(res.conn);
            }
        } catch {
            //
        }
    }

    public render() {
        const { styles } = this.props;

        return (
            <View style={styles.container}>
                <View style={styles.topContainer} testID="welcome-text">
                    <Text large style={styles.welcomeTitle}>
                        {translate('Onboarding.welcomeTitle')}
                    </Text>
                    <Text style={styles.welcomeTextWeb}>
                        {translate('Onboarding.welcomeTextWeb')}
                    </Text>
                </View>

                <View style={styles.canvas}>
                    <canvas ref={ref => (this.qrCanvas = ref)} style={{ textAlign: 'center' }} />
                </View>

                <View style={styles.bottomContainer}>
                    <View style={styles.infoRow}>
                        <View key="circle-1" style={styles.circle}>
                            <Text key="number-1" style={styles.number}>{`1`}</Text>
                        </View>
                        <Text style={styles.text}>{translate('Onboarding.webStep1')}</Text>
                    </View>

                    <View style={styles.infoRow}>
                        <View key="circle-2" style={styles.circle}>
                            <Text key="number-2" style={styles.number}>{`2`}</Text>
                        </View>
                        <Text style={styles.text}>{translate('Onboarding.webStep2')}</Text>
                    </View>

                    <View style={styles.infoRow}>
                        <View key="circle-3" style={styles.circle}>
                            <Text key="number-3" style={styles.number}>{`3`}</Text>
                        </View>
                        <Text style={styles.text}>{translate('Onboarding.webStep3')}</Text>
                    </View>
                </View>
            </View>
        );
    }
}

export const OnboardingScreen = smartConnect(OnboardingScreenComponent, [
    withTheme(stylesProvider),
    withNavigationParams()
]);
