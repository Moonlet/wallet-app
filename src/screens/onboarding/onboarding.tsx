import React from 'react';
import { View, Image } from 'react-native';
import { Text, Button } from '../../library';
import {
    NavigationParams,
    NavigationScreenProp,
    NavigationState,
    NavigationActions,
    StackActions
} from 'react-navigation';
import stylesProvider from './styles';
import { withTheme } from '../../core/theme/with-theme';
import { createHDWallet } from '../../redux/wallets/actions';
import { connect } from 'react-redux';
import { smartConnect } from '../../core/utils/smart-connect';
import { hash } from '../../core/secure/encrypt';
import { setPassword } from '../../core/secure/keychain';
import { translate } from '../../core/i18n';
import { isFeatureActive, RemoteFeature } from '../../core/utils/remote-feature-config';
import { openLoadingModal } from '../../redux/ui/loading-modal/actions';

export interface IProps {
    navigation: NavigationScreenProp<NavigationState, NavigationParams>;
    styles: ReturnType<typeof stylesProvider>;
}

export interface IReduxProps {
    createHDWallet: typeof createHDWallet;
    openLoadingModal: typeof openLoadingModal;
}

const mapDispatchToProps = {
    createHDWallet,
    openLoadingModal
};

const navigationOptions = () => ({ header: null });

export class OnboardingScreenComponent extends React.Component<IProps & IReduxProps> {
    public static navigationOptions = navigationOptions;

    public mnemonic = [
        'pigeon',
        'road',
        'portion',
        'echo',
        'robust',
        'panel',
        'fat',
        'dune',
        'direct',
        'flee',
        'pave',
        'tonight',
        'govern',
        'blue',
        'tree',
        'actual',
        'palace',
        'rude',
        'legal',
        'sand',
        'width',
        'clever',
        'order',
        'icon'
    ];

    public onPressRecover() {
        this.props.navigation.navigate('RecoverWallet');
    }
    public onPressCreate() {
        this.props.navigation.navigate('CreateWalletMnemonic');
    }

    public onPressConnect() {
        this.props.navigation.navigate('ConnectHardwareWallet');
    }

    public async onPressGenerateWallet() {
        this.props.openLoadingModal();
        const password = await hash('000000');
        setPassword(password, false);
        this.props.createHDWallet(this.mnemonic.join(' '), password, () => {
            this.props.navigation.dispatch(StackActions.popToTop());
            this.props.navigation.navigate(
                'MainNavigation',
                {},
                NavigationActions.navigate({ routeName: 'Dashboard' })
            );
        });
    }

    public render() {
        const styles = this.props.styles;

        return (
            <View style={styles.container}>
                <View style={styles.topContainer}>
                    <Image
                        style={styles.logoImage}
                        source={require('../../assets/images/png/moonlet_space.png')}
                    />

                    <View style={styles.textContainer}>
                        <Text style={styles.welcomeTitle}>
                            {translate('Onboarding.welcomeTitle')}
                        </Text>
                        <Text style={styles.welcomeText}>
                            {translate('Onboarding.welcomeText')}
                        </Text>
                    </View>
                </View>

                <View style={styles.buttonsContainer}>
                    <View style={styles.topButtons}>
                        <Button
                            style={[styles.button, styles.leftButton]}
                            onPress={() => this.onPressRecover()}
                            testID="button-recover"
                        >
                            {translate('App.labels.recover')}
                        </Button>
                        <Button
                            style={[styles.button, styles.rightButton]}
                            onPress={() => this.onPressConnect()}
                        >
                            {translate('App.labels.connect')}
                        </Button>
                    </View>

                    <Button
                        testID="button-create"
                        style={styles.bottomButton}
                        primary
                        onPress={() => this.onPressCreate()}
                    >
                        {translate('App.labels.create')}
                    </Button>

                    {isFeatureActive(RemoteFeature.DEV_TOOLS) && (
                        <Button
                            testID="button-generate"
                            style={styles.bottomButton}
                            primary
                            onPress={() => this.onPressGenerateWallet()}
                        >
                            {`Generate wallet`}
                        </Button>
                    )}
                </View>
            </View>
        );
    }
}

export const OnboardingScreen = smartConnect(OnboardingScreenComponent, [
    connect(null, mapDispatchToProps),
    withTheme(stylesProvider)
]);
