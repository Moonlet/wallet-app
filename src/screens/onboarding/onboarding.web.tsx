import React from 'react';
import { View } from 'react-native';
import { Text } from '../../library';
import { NavigationActions } from 'react-navigation';
import stylesProvider from './styles-web';
import { withTheme, IThemeProps } from '../../core/theme/with-theme';
import { smartConnect } from '../../core/utils/smart-connect';
import { translate } from '../../core/i18n';
import { withNavigationParams, INavigationProps } from '../../navigation/with-navigation-params';
import QRCode from 'qrcode';
import { WalletConnectWeb } from '../../core/wallet-connect/wallet-connect-web';

export class OnboardingScreenComponent extends React.Component<
    INavigationProps & IThemeProps<ReturnType<typeof stylesProvider>>
> {
    public qrCanvas: HTMLCanvasElement;
    public unsubscribe: () => void;

    public componentDidMount() {
        if (!WalletConnectWeb.isConnected()) {
            WalletConnectWeb.connect().then(uri => {
                // tslint:disable-next-line
                console.log(uri);

                QRCode.toCanvas(this.qrCanvas, uri, { errorCorrectionLevel: 'H' });
                this.unsubscribe = WalletConnectWeb.subscribe('connect', payload => {
                    WalletConnectWeb.getState().then(() => {
                        this.props.navigation.navigate(
                            'MainNavigation',
                            {},
                            NavigationActions.navigate({ routeName: 'Dashboard' })
                        );
                    });
                });
            });
        }
    }

    public componentWillUnmount() {
        if (this.unsubscribe) {
            this.unsubscribe();
        }
    }

    public onPressDisconnect() {
        WalletConnectWeb.disconnect();
    }

    public render() {
        const { styles } = this.props;

        return (
            <View style={styles.container}>
                <View style={styles.topContainer} testID="welcome-text">
                    <Text large style={styles.welcomeTitle}>
                        {translate('Onboarding.welcomeTitle')}
                    </Text>
                    <Text style={styles.welcomeTextWeb}>
                        {translate('Onboarding.welcomeTextWeb')}
                    </Text>
                </View>

                <View style={styles.canvas}>
                    <canvas ref={ref => (this.qrCanvas = ref)} style={{ textAlign: 'center' }} />
                </View>

                <View style={styles.bottomContainer}>
                    <View style={styles.infoRow}>
                        <View key="circle-1" style={styles.circle}>
                            <Text key="number-1" style={styles.number}>{`1`}</Text>
                        </View>
                        <Text style={styles.text}>{translate('Onboarding.webStep1')}</Text>
                    </View>

                    <View style={styles.infoRow}>
                        <View key="circle-2" style={styles.circle}>
                            <Text key="number-2" style={styles.number}>{`2`}</Text>
                        </View>
                        <Text style={styles.text}>{translate('Onboarding.webStep2')}</Text>
                    </View>

                    <View style={styles.infoRow}>
                        <View key="circle-3" style={styles.circle}>
                            <Text key="number-3" style={styles.number}>{`3`}</Text>
                        </View>
                        <Text style={styles.text}>{translate('Onboarding.webStep3')}</Text>
                    </View>
                </View>
            </View>
        );
    }
}

export const OnboardingScreen = smartConnect(OnboardingScreenComponent, [
    withTheme(stylesProvider),
    withNavigationParams()
]);
