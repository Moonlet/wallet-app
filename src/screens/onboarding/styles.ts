import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { pw } from '../../styles';
import { BASE_DIMENSION, normalize } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            flex: 1,
            backgroundColor: theme.colors.appBackground
        },
        topContainer: {
            flex: 1,
            justifyContent: 'center',
            marginBottom: BASE_DIMENSION * 4
        },
        buttonsContainer: {
            paddingHorizontal: BASE_DIMENSION * 4,
            marginBottom: BASE_DIMENSION * 7
        },
        topButtons: {
            flexDirection: 'row',
            marginBottom: BASE_DIMENSION * 2
        },
        leftButton: {
            marginRight: BASE_DIMENSION / 2
        },
        rightButton: {
            marginLeft: BASE_DIMENSION / 2
        },
        bottomButton: {
            marginBottom: BASE_DIMENSION * 2
        },
        logoImage: {
            width: pw(60),
            resizeMode: 'contain',
            alignSelf: 'center'
        },
        textContainer: {
            marginHorizontal: BASE_DIMENSION * 4
        },
        welcomeTitle: {
            fontSize: normalize(22),
            lineHeight: normalize(28),
            fontWeight: 'bold',
            textAlign: 'center',
            letterSpacing: 0.35,
            color: theme.colors.text,
            marginBottom: BASE_DIMENSION
        },
        welcomeText: {
            fontSize: normalize(17),
            lineHeight: normalize(22),
            textAlign: 'center',
            color: theme.colors.textSecondary
        }
    });
