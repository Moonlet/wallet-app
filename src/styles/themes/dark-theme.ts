import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
