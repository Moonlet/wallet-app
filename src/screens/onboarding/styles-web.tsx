import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { pw } from '../../styles';
import { BASE_DIMENSION } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            flex: 1,
            backgroundColor: theme.colors.appBackground
        },
        topContainer: {
            flex: 1,
            justifyContent: 'center'
        },
        bottomContainer: {
            flex: 1,
            justifyContent: 'center',
            paddingHorizontal: BASE_DIMENSION * 3,
            marginTop: BASE_DIMENSION * 2
        },
        logoImage: {
            width: pw(60),
            resizeMode: 'contain',
            alignSelf: 'center'
        },
        welcomeTitle: {
            fontSize: 22,
            lineHeight: 28,
            fontWeight: 'bold',
            textAlign: 'center',
            letterSpacing: 0.35,
            color: theme.colors.text,
            marginBottom: BASE_DIMENSION
        },
        infoRow: {
            flexDirection: 'row',
            marginBottom: BASE_DIMENSION * 2
        },
        circle: {
            height: 30,
            width: 30,
            borderRadius: 15,
            backgroundColor: theme.colors.accentSecondary,
            marginRight: BASE_DIMENSION * 2
        },
        number: {
            fontSize: 17,
            lineHeight: 21,
            textAlign: 'center',
            color: theme.colors.accent,
            alignSelf: 'center',
            marginTop: BASE_DIMENSION / 2
        },
        welcomeTextWeb: {
            fontSize: 16,
            lineHeight: 21,
            textAlign: 'center',
            color: theme.colors.textSecondary,
            paddingHorizontal: BASE_DIMENSION * 4
        },
        text: {
            fontSize: 15,
            lineHeight: 19,
            color: theme.colors.text,
            marginTop: BASE_DIMENSION / 2
        },
        canvas: {
            justifyContent: 'center',
            alignContent: 'center',
            alignItems: 'center'
        }
    });
