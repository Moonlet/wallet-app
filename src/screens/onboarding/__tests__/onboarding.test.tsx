import React from 'react';
import { OnboardingScreenComponent, IProps } from '../onboarding';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';
import { loadTranslations } from '../../../core/i18n';

const props: IProps = {
    // @ts-ignore
    navigation: {
        navigate: jest.fn()
    },
    styles: styleProvider(darkTheme)
};

describe('creat wallet terms screen component', () => {
    beforeAll(async () => {
        await loadTranslations('en');
    });
    it('renders correctly', () => {
        const wrapper = shallow(<OnboardingScreenComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('navigates to create screen', () => {
        const wrapper = shallow(<OnboardingScreenComponent {...props} />);
        wrapper.find('[testID="button-create"]').simulate('Press');
        expect(props.navigation.navigate).toHaveBeenCalledWith(
            'CreateWalletNavigation',
            {},
            'navigate-action'
        );
    });
});
