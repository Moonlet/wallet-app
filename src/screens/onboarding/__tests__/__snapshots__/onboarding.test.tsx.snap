// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creat wallet terms screen component renders correctly 1`] = `
"<View style={{...}}>
  <View style={{...}}>
    <View style={{...}}>
      <Image style={{...}} source={{...}} />
    </View>
    <View style={{...}}>
      <withTheme(TextComponent) large={true} style={{...}}>
        Welcome to Moonlet!
      </withTheme(TextComponent)>
      <withTheme(TextComponent) style={{...}} darker={true}>
        Weâ€™ll generate this section once you create, recover or connect a wallet
      </withTheme(TextComponent)>
    </View>
  </View>
  <View style={{...}}>
    <View style={{...}}>
      <withTheme(ButtonComponent) style={{...}} onPress={[Function: onPress]} testID=\\"button-recover\\">
        Recover
      </withTheme(ButtonComponent)>
      <withTheme(ButtonComponent) style={{...}} onPress={[Function: onPress]}>
        Connect
      </withTheme(ButtonComponent)>
    </View>
    <View style={{...}}>
      <withTheme(ButtonComponent) testID=\\"button-create\\" style={{...}} primary={true} onPress={[Function: onPress]}>
        Create
      </withTheme(ButtonComponent)>
    </View>
    <View style={{...}}>
      <withTheme(ButtonComponent) testID=\\"button-generate\\" style={{...}} primary={true} onPress={[Function: onPress]}>
        Generate wallet
      </withTheme(ButtonComponent)>
    </View>
  </View>
</View>"
`;
